### PR TITLE
treewide/nixos: remove `with lib;` part 2

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #208242

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/misc/dysnomia.nix [hunk 1]
```diff
@@ -84,68 +81,68 @@ in
...
         description = "An attribute set in which each key represents a container and each value an attribute set providing its configuration properties";
         default = {};
-        type = types.attrsOf types.attrs;
+        type = lib.types.attrsOf lib.types.attrs;
       };
 
-      components = mkOption {
-        description = "An attribute set in which each key represents a container and each value an attribute set in which each key represents a component and each value a derivation constructing its initial state";
+      components = lib.mkOption {
+        description = "An attribute set in which each key represents a container and each value an attribute set in which each key represents a component and each value a derivation lib.constructing its initial state";
```
Comment (philiptaron) [inline nixos/modules/services/misc/dysnomia.nix:114]: revert

File: nixos/modules/services/networking/kresd.nix [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
-
-with lib;
-
 let
   cfg = config.services.kresd;
 
   # Convert systemd-style address specification to kresd config line(s).
   # On Nix level we don't attempt to precisely validate the address specifications.
-  # The optional IPv6 scope spec comes *after* port, perhaps surprisingly.
+  # The lib.optional IPv6 scope spec comes *after* port, perhaps surprisingly.
```
Comment (philiptaron) [inline nixos/modules/services/networking/kresd.nix:7]: revert

File: nixos/modules/system/boot/systemd/journald.nix [hunk 3]
```diff
@@ -1,49 +1,46 @@
...
       default = "30s";
-      type = types.str;
+      type = lib.types.str;
       description = ''
         Configures the rate limiting interval that is applied to all
         messages generated on the system. This rate limiting is applied
         per-service, so that two services which log do not interfere with
         each other's limit. The value may be specified in the following
-        units: s, min, h, ms, us. To turn off any kind of rate limiting,
+        units: s, min, h, ms, us. To turn off lib.any kind of rate limiting,
```
Comment (philiptaron) [inline nixos/modules/system/boot/systemd/journald.nix:24]: revert

**Review by philiptaron — CHANGES_REQUESTED** [review]
I was only able to find a few defects in this batch.

File: nixos/modules/services/matrix/appservice-discord.nix [hunk 4]
```diff
@@ -13,14 +10,14 @@ let
...
 
-      package = mkPackageOption pkgs "matrix-appservice-discord" { };
+      package = lib.mkPackageOption pkgs "matrix-appservice-discord" { };
 
-      settings = mkOption rec {
-        # TODO: switch to types.config.json as prescribed by RFC42 once it's implemented
-        type = types.attrs;
-        apply = recursiveUpdate default;
+      settings = lib.mkOption rec {
+        # TODO: switch to lib.types.config.json as prescribed by RFC42 once it's implemented
```
[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply nixos/modules/services/matrix/appservice-discord.nix:18]: what do you suggest here?

[the remark this answers is not in the document]
Comment (philiptaron) [reply nixos/modules/services/matrix/appservice-discord.nix:18]: Nothing. I was learning as I read. I hope you don't mind.

**Review by philiptaron — APPROVED** [review]

File: nixos/modules/services/matrix/appservice-discord.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply nixos/modules/services/matrix/appservice-discord.nix:18]: the fire symbol was not clear to me :laughing: . all good

**Comment by eclairevoyant** [comment]
Also `lib.optionalFile` is not a function that exists, how did it get caught up in this treewide?

**Comment by eclairevoyant** [comment]
Both are issues. I'm asking for the script so we know the magnitude of the mistake.

**Comment by philiptaron** [comment]
> Both are issues. I'm asking for the script so we know the magnitude of the mistake.

I asked Felix (@Stunkymonkey) for the same thing and received this: https://github.com/NixOS/nixpkgs/pull/335603#issuecomment-2303044418

**Comment by eclairevoyant** [comment]
Thanks, I'll continue the conversation on that PR then.

**Pull request merged**